### PR TITLE
Issue 1181

### PR DESCRIPTION
**Pull request by q0w**
# I have made things!

## Checklist

- [ ] I have double checked that there are no unrelated changes in this pull request (old patches, accidental config files, etc)
- [ ] I have created at least one test case for the changes I have made
- [ ] I have updated the documentation for the changes I have made
- [ ] I have added my changes to the `CHANGELOG.md`

## Related issues
#1181 

🙏 Please, if you or your company finds `dry-python` valuable, help us sustain the project by sponsoring it transparently on https://github.com/sponsors/dry-python. As a thank you, your profile/company logo will be added to our main README which receives hundreds of unique visitors per day.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -65,7 +65,7 @@ trio = "^0.19"
 attrs = "^21.2"
 httpx = "^0.21"
 
-mypy = "^0.910"
+mypy = "0.920"
```
Comment (sobolevn) [inline pyproject.toml:68]:
```suggestion
mypy = "^0.920"
```

We don't pin `mypy`, this is a dev-only dep.

File: typesafety/test_interfaces/test_specific/test_ioresult/test_ioresultlike_inheritance.yml [hunk 2]
```diff
@@ -159,7 +159,16 @@
     main:18: note: See https://mypy.readthedocs.io/en/stable/common_issues.html#incompatible-overrides
     main:18: note: This violates the Liskov substitution principle
     main:27: error: Signature of "from_ioresult" incompatible with supertype "IOResultLikeN"
+    main:27: note:          def [_NewValueType, _NewErrorType] from_ioresult(self, inner_value: IOResult[_NewValueType, _NewErrorType]) -> MyClass[_NewValueType, _NewErrorType]
+    main:27: note:          def [_ValueType, _ErrorType] from_ioresult(cls, inner_value: IOResult[_ValueType, _ErrorType]) -> KindN[MyClass[_ValueType, _ErrorType], _ValueType, _ErrorType, NoReturn]
+    main:27: note:      Subclass:
```
Comment (sobolevn) [inline typesafety/test_interfaces/test_specific/test_ioresult/test_ioresultlike_inheritance.yml:164]: Can you please also bump `pytest-mypy-plugins`? This error will go away (I mean invalid ordering).

**Review by sobolevn — CHANGES_REQUESTED** [review]
Thanks! Much appreciated! 👍

File: typesafety/test_interfaces/test_specific/test_ioresult/test_ioresultlike_inheritance.yml (hunk 2 above)
Comment (q0w) [reply typesafety/test_interfaces/test_specific/test_ioresult/test_ioresultlike_inheritance.yml:164]: sure

**Review by sobolevn — APPROVED** [review]
Oh my, 48 files! This is huge amount of work!
Thank you so much, I really appreciate your time and effort! 🤝

**Pull request merged**